bindings/rust-examples/session-resumption/src/bin/client.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls::{
+    callbacks::{SessionTicket, SessionTicketCallback},
+    connection::{Connection, ModifiedBuilder},
+    security::DEFAULT_TLS13,
+};
+use s2n_tls_tokio::TlsConnector;
+use std::{
+    collections::HashMap,
+    error::Error,
+    net::IpAddr,
+    sync::{Arc, Mutex},
+};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+};
+
+struct ApplicationContext {
```

Comment:
Nit, in this case, we are the application so this naming feels a bit odd. `ConnectionContext`, perhaps?